### DISCUSSION TITLE
how to set a static json file in place of generated ones

### DISCUSSION THREAD

**Discussion by DennyFiguerres** [RicoSuter/NSwag#4597] [Q&A]
i have odata apis.  i have tools that generate an open api 3 json file.
how do i tell nswag to use my file and not the default.

**Comment by DennyFiguerres** [comment]
Still seeking this, many googles have not found an answer. cant find it on stack overflow,,,,,
really stuck here.... i hope there is a simple answer that i am just not seeing.

**Comment by lahma** [comment]
Maybe something like this in `nswag.json` could work?

```json
{
  "runtime": "Net80",
  "defaultVariables": null,
  "documentGenerator": {
    "fromDocument": {
      "url": "c:\\temp\\rest-api.yaml",
      "output": "C:\\temp\\swagger-temp.json",
      "newLineBehavior": "Auto"
    }
  },
  "codeGenerators": { 
   // add generators here
  }
}
```

**Reply by DennyFiguerres** [reply]
first thank you for the reply!

i do not have a file like you just named so let make sure where this is and what i need

so i have a vs project that when it runs it has the url:   https://localhost/swagger/index.html
and the nswag / swagger ui wants the get the swagger / openapi json from the url: 
https://localhost/swagger/v1/swagger.json

but in my case i have a file already that is in the static web files that i need the swagger ui to use in place of the auto generated one.
so i want to replace 

https://localhost/swagger/v1/swagger.json
with
https://localhost/myopenapi.json

so that the swagger web page uses my file to show the api's i have.

is that what you are showing me a way to do ?

**Reply by vladimir-bukreev** [reply]
How to apply this solution to an ASP.NET Core application? I don't see the ability to configure document generators, and I don't understand how to use the FromDocumentCommand type.
